app/assets/src/components/views/compare/SamplesHeatmapView/SamplesHeatmapView.jsx

```diff
@@ -512,6 +511,53 @@ class SamplesHeatmapView extends React.Component {
     return true;
   }
 
+  getTopTaxaPerSample(filteredTaxonIds) {
+    // Fetch the top N taxa from each sample, sorted by the selected metric,
+    // that passed all selected filters.
+    let {
+      sampleDetails,
+      allData,
+      allTaxonDetails,
+      selectedOptions,
+    } = this.state;
+    let { metric, taxonsPerSample } = selectedOptions;
+    let topTaxIds = new Set(),
+      topTaxonDetails = {},
+      filteredData = {};
+
+    Object.values(sampleDetails).forEach(sample => {
+      let filteredTaxaInSample = sample.taxa.filter(taxon =>
+        filteredTaxonIds.has(taxon)
+      );
+
+      filteredTaxaInSample.sort(
+        (taxId1, taxId2) =>
+          allData[metric][allTaxonDetails[taxId2].index][sample.index] -
+          allData[metric][allTaxonDetails[taxId1].index][sample.index]
+      );
+
+      let count = 0;
+      for (let taxId of filteredTaxaInSample) {
+        if (count >= taxonsPerSample) {
+          break;
+        } else if (!topTaxIds.has(taxId)) {
+          let taxon = allTaxonDetails[taxId];
+          topTaxIds.add(taxId);
+          topTaxonDetails[taxId] = allTaxonDetails[taxId];
+          topTaxonDetails[taxon["name"]] = allTaxonDetails[taxId];
+
+          this.props.metrics.forEach(metric => {
```

Comment:
Please load props on top of the function just like you loaded state, i.e. `const { metrics } = this.props`

Reply:
Done!